src/System.Text.Json/src/System/Text/Json/Serialization/JsonEnumerableConverter.cs

```diff
@@ -8,6 +8,32 @@ namespace System.Text.Json.Serialization.Policies
 {
     internal abstract class JsonEnumerableConverter
     {
-        public abstract IEnumerable CreateFromList(Type elementType, IList sourceList);
+        public abstract IEnumerable CreateFromList(Type enumerableType, Type elementType, IList sourceList);
+
+        protected static Array CreateArrayFromList(Type elementType, IList sourceList)
+        {
+            Array array;
+
+            if (sourceList.Count > 0 && sourceList[0] is Array probe)
+            {
+                array = Array.CreateInstance(probe.GetType(), sourceList.Count);
+
+                int i = 0;
+                foreach (IList child in sourceList)
+                {
+                    if (child is Array childArray)
+                    {
+                        array.SetValue(childArray, i++);
+                    }
+                }
```

Comment:
What's this case intended to handle? Why does this need different handling?

Reply:
This code supports the `DefaultArrayConverter` and handles 1-D arrays I believe. cc @steveharter